source/FireCommand.cpp

```diff
@@ -0,0 +1,104 @@
+/* FireCommand.cpp
+Copyright (c) 2021 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "FireCommand.h"
+
+#include <cmath>
+#include <vector>
+
+using namespace std;
+
+namespace {
+	template <typename T>
+	void SubsetAssign(std::vector<T> &lhs, const std::vector<T> &rhs)
+	{
+		const auto size = lhs.size() < rhs.size() ? lhs.size() : rhs.size();
+		for(size_t i = 0; i < size; ++i)
+			lhs[i] = rhs[i];
+	}
+}
+
+
+
+// Sets the specified amount of hardpoints desired.
+void FireCommand::SetHardpoints(size_t count)
+{
+	weapon.Resize(count);
+	aim.resize(count);
+}
+
+
+
+// Assigns the subset of other to this class that is no larger than
+// this command's hardpoint size.
+void FireCommand::AssignSubsetOf(const FireCommand &other)
+{
+	SubsetAssign(weapon.Bits(), other.weapon.Bits());
+	SubsetAssign(aim, other.aim);
+}
+
+
+
+// Reset this to an empty command.
+void FireCommand::Clear()
+{
+	weapon.Clear();
+	aim.clear();
```

Comment:
Don't they both have no bits at all right now? Since clear sets the size to 0. Also, we'd want to keep the current "reserved" buffer for both to avoid heap thrashing.